src/content_mapblock.cpp

```diff
@@ -525,1633 +371,955 @@ static TileSpec getSpecialTile(const ContentFeatures &f,
-							 visible_faces[5] ? -b : -a + offset,
-							 visible_faces[2] ? b : a - offset,
-							 visible_faces[0] ? b * vlev : a * vlev - offset,
-							 visible_faces[4] ? b : a - offset);
-				makeAutoLightedCuboid(&collector, data, pos, box, tile, special_color, frame);
-			}
-		break;}
-		case NDT_ALLFACES:
-		{
-			TileSpec tile_leaves = getNodeTile(n, p,
-					v3s16(0,0,0), data);
-			u16 l = getInteriorLight(n, 1, nodedef);
-			video::SColor c = encode_light_and_color(l,
-				tile_leaves.color, f.light_source);
-
-			v3f pos = intToFloat(p, BS);
-			aabb3f box(-BS/2,-BS/2,-BS/2,BS/2,BS/2,BS/2);
-			makeAutoLightedCuboid(&collector, data, pos, box, tile_leaves, c, frame);
-		break;}
-		case NDT_ALLFACES_OPTIONAL:
-			// This is always pre-converted to something else
-			FATAL_ERROR("NDT_ALLFACES_OPTIONAL not pre-converted");
-			break;
-		case NDT_TORCHLIKE:
-		{
-			v3s16 dir = n.getWallMountedDir(nodedef);
-
-			u8 tileindex = 0;
-			if(dir == v3s16(0,-1,0)){
-				tileindex = 0; // floor
-			} else if(dir == v3s16(0,1,0)){
-				tileindex = 1; // ceiling
-			// For backwards compatibility
-			} else if(dir == v3s16(0,0,0)){
-				tileindex = 0; // floor
-			} else {
-				tileindex = 2; // side
-			}
+	u8 param2 = n.getParam2();
+	bool H_merge = !(param2 & 128);
+	bool V_merge = !(param2 & 64);
+	param2 &= 63;
+
+	static const float a = BS / 2;
+	static const float g = a - 0.003;
+	static const float b = .876 * ( BS / 2 );
+
+	static const aabb3f frame_edges[12] = {
+		aabb3f( b, b,-a, a, a, a), // y+
+		aabb3f(-a, b,-a,-b, a, a), // y+
+		aabb3f( b,-a,-a, a,-b, a), // y-
+		aabb3f(-a,-a,-a,-b,-b, a), // y-
+		aabb3f( b,-a, b, a, a, a), // x+
+		aabb3f( b,-a,-a, a, a,-b), // x+
+		aabb3f(-a,-a, b,-b, a, a), // x-
+		aabb3f(-a,-a,-a,-b, a,-b), // x-
+		aabb3f(-a, b, b, a, a, a), // z+
+		aabb3f(-a,-a, b, a,-b, a), // z+
+		aabb3f(-a,-a,-a, a,-b,-b), // z-
+		aabb3f(-a, b,-a, a, a,-b), // z-
+	};
+	static const aabb3f glass_faces[6] = {
+		aabb3f(-g,-g, g, g, g, g), // z+
+		aabb3f(-g, g,-g, g, g, g), // y+
+		aabb3f( g,-g,-g, g, g, g), // x+
+		aabb3f(-g,-g,-g, g, g,-g), // z-
+		aabb3f(-g,-g,-g, g,-g, g), // y-
+		aabb3f(-g,-g,-g,-g, g, g), // x-
+	};
 
-			TileSpec tile = getNodeTileN(n, p, tileindex, data);
-			tile.material_flags &= ~MATERIAL_FLAG_BACKFACE_CULLING;
-			tile.material_flags |= MATERIAL_FLAG_CRACK_OVERLAY;
-
-			u16 l = getInteriorLight(n, 1, nodedef);
-			video::SColor c = encode_light_and_color(l, tile.color,
-				f.light_source);
-
-			float s = BS/2*f.visual_scale;
-			// Wall at X+ of node
-			video::S3DVertex vertices[4] =
-			{
-				video::S3DVertex(-s,-s,0, 0,0,0, c, 0,1),
-				video::S3DVertex( s,-s,0, 0,0,0, c, 1,1),
-				video::S3DVertex( s, s,0, 0,0,0, c, 1,0),
-				video::S3DVertex(-s, s,0, 0,0,0, c, 0,0),
-			};
+	// tables of neighbour (connect if same type and merge allowed),
+	// checked with g_26dirs
+
+	// 1 = connect, 0 = face visible
+	bool nb[18] = { 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 };
```

Comment:
how to name it? it’s count of 6d offsets in one or two coordinates. perhaps smth like `D6D_COUNT_2`?